app/src/js/components/Rules/overview.js

```diff
@@ -27,86 +37,128 @@ const breadcrumbConfig = [
   },
 ];
 
-class RulesOverview extends React.Component {
-  constructor() {
-    super();
-    this.generateBulkActions = this.generateBulkActions.bind(this);
-  }
+const RulesOverview = ({ urlHelper }) => {
+  const dispatch = useDispatch();
+  const location = useLocation();
+  const { queryParams, getPersistentQueryParams } = urlHelper;
+
+  const rules = useSelector((state) => state.rules);
+  const { list } = rules || {};
+  const { count, queriedAt } = list?.meta || {};
+
+  const removeEsFields = (data) => {
+    const { queriedAt: _, timestamp, stats, ...nonEsFields } = data;
+    return nonEsFields;
+  };
 
-  componentDidMount() {
-    this.props.dispatch(listRules);
-  }
+  useEffect(() => {
+    dispatch(listRules());
+  }, [dispatch]);
 
-  generateBulkActions() {
-    const { rules } = this.props;
-    return bulkActions(rules);
-  }
+  const generateBulkActions = useCallback(() => {
+    if (!rules || !rules.list || !rules.list.data) {
+      return [];
+    }
 
-  render() {
-    const { queryParams, rules } = this.props;
-    const { list } = rules;
-    const { count, queriedAt } = list.meta;
-    return (
-      <div className="page__component">
-        <Helmet>
-          <title> Rules Overview </title>
-        </Helmet>
-        <section className="page__section page__section__controls">
-          <Breadcrumbs config={breadcrumbConfig} />
-        </section>
-        <section className="page__section page__section__header-wrapper">
-          <div className="page__section__header">
-            <h1 className="heading--large heading--shared-content with-description">
-              Rules Overview
-            </h1>
-            {lastUpdated(queriedAt)}
-          </div>
-        </section>
-        <section className="page__section">
-          <div className="heading__wrapper--border">
-            <h2 className="heading--medium heading--shared-content with-description">
-              All Rules
-              <span className="num-title">
-                {count ? ` ${tally(count)}` : 0}
-              </span>
-            </h2>
-          </div>
+    const rulesOps = {
+      list: rules.list,
+      enabled: rules.enabled || {},
```

Comment:
Is an object the correct type to assign if rules.enabled is false? Shouldn't it be `false`? Same question for the next two...